dcos_test_utils/gce.py

```diff
@@ -0,0 +1,276 @@
+           resource,see https://cloud.google.com/compute/docs/reference/latest/instances"""
+        response = self.compute.instances().get(project=self.project_id, zone=zone, instance=name).execute()
+        log.debug('get_instance_info response: ' + str(response))
+        return response
+
+    @catch_http_exceptions
+    def list_group_instances(self, group_name, zone) -> typing.Iterator(dict):
+        response = self.compute.instanceGroupManagers().listManagedInstances(project=self.project_id, zone=zone,
+                                                                             instanceGroupManager=group_name).execute()
+        log.debug('list_group_instances response: ' + str(response))
+
+        for instance in response['managedInstances']:
+            yield instance
+
+    @retry(wait_fixed=2000, retry_on_result=lambda res: res is None, stop_max_delay=30 * 1000)
+    def get_instance_network_properties(self, instance_name, zone) -> dict:
+        network_info = self.get_instance_info(instance_name, zone)['networkInterfaces'][0]
+        if 'networkIP' not in network_info or 'accessConfigs' not in network_info:
+            return None
+        if 'natIP' not in network_info['accessConfigs'][0]:
+            return None
+        return network_info
+
+    @catch_http_exceptions
+    def create_deployment(self, name, deployment_config):
+        body = {
+            'name': name,
+            'target': {
+                'config': {
+                    'content': yaml.dump(deployment_config, default_flow_style=False)}
+            }
+        }
+
+        response = self.deployment_manager.deployments().insert(
+            project=self.project_id, body=body).execute()
+        log.debug('create_deployment response: ' + str(response))
+
+
+class Deployment:
+    def __init__(self, gce_wrapper, name, zone):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.zone = zone
+
+    @catch_http_exceptions
+    def delete(self):
+        response = self.gce_wrapper.deployment_manager.deployments().delete(project=self.gce_wrapper.project_id,
+                                                                            deployment=self.name).execute()
+        log.debug('delete_deployment response: ' + str(response))
+
+    @catch_http_exceptions
+    def get_info(self) -> dict:
+        """Returns the dictionary representation of a GCE deployment resource. For details on the contents of this
+           resource, see https://cloud.google.com/deployment-manager/docs/reference/latest/deployments#resource"""
+        response = self.gce_wrapper.deployment_manager.deployments().get(project=self.gce_wrapper.project_id,
+                                                                         deployment=self.name).execute()
+        log.debug('get_deployment_info response: ' + str(response))
+        return response
+
+    def _check_status(response: dict) -> bool:
```

Comment:
Will break - making it static means the retry decorator calling it needs to specify Deployment._check_status instead of only _check_status, but this is not possible since for some reason, the retry decorator has no knowledge of what the Deployment class is